api/python/cellxgene_census/src/cellxgene_census/_release_directory.py

```diff
@@ -35,9 +40,31 @@
 )
 CensusDirectory = Dict[CensusVersionName, Union[CensusVersionName, CensusVersionDescription]]
 
+CensusMirrorName = str  # name of the mirror
+CensusMirror = TypedDict(
```

Comment:
The fields need better comments and typing.  Reading the code, I can't quite tell what the legal values are, or how they will be used.  I recommend:
* Use `Literal` for `provider` (i.e., for any enumeration)
* Document the semantics / constraints of each field, if any.  E.g., `base_uri` must be absolute, no?